### PR TITLE
Load cross sections into Processes

### PR DESCRIPTION
**Pull request by sethrj**
This directly couples the cross sections for Livermore and Compton processes to imported data and provides the interface for adding additional processes with ease.

I needed to add several additions/changes to helper functions for implementation and testing -- I've broken out these changes and descriptions into the commit history. I'd be happy to break any or all of those out into one or more separate PRs if they're not relevant enough to this PR.

## ROOT data update

I had to update the exported process cross sections with a new version of geant4 because it looks like Geant4 10.6 generates nonuniform log grids in energy. Here is the old Compton cross section data (stainless steel)
stupid-geant

and the new cross sections:
less-stupid-geant

### CODE REVIEW

File: src/physics/grid/ValueGridBuilder.cc [hunk 1]
```diff
@@ -155,6 +189,24 @@ auto ValueGridXsBuilder::build(ValueGridInserter insert) const -> ValueGridId
 
 //---------------------------------------------------------------------------//
 // LOG BUILDER
+//---------------------------------------------------------------------------//
+/*!
+ * Construct XS arrays from scaled (*E) data from Geant4.
```
Comment (amandalund) [inline src/physics/grid/ValueGridBuilder.cc:194]: Description doesn't match the function.

File: src/physics/base/ImportedProcessAdapter.cc [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                           "Duplicate table type '"
+                               << to_cstring(tables[table_id.get()].table_type)
+                               << "' in process data for '"
+                               << to_cstring(process_class) << "'");
+            *dst = table_id;
+        }
+
+        auto particle_id = particles->find(pdg);
+        CELER_VALIDATE(particle_id,
+                       "Parrticle PDG{" << pdg.get() << "} is not available");
```
Comment (amandalund) [inline src/physics/base/ImportedProcessAdapter.cc:108]:
```suggestion
                       "Particle PDG{" << pdg.get() << "} is not available");
```

**Review by amandalund — APPROVED** [review]
Very nice @sethrj!

File: src/physics/base/ImportedProcessAdapter.cc (hunk 2 above)
Comment (sethrj) [reply src/physics/base/ImportedProcessAdapter.cc:108]: My poor macbook's keys have been super sticky lately :(

Comment (mrguilima) [reply src/physics/base/ImportedProcessAdapter.cc:108]: You know what?  I had the very same 'r' key sticking in my laptop.  There is a known problem with mac keyboards, and I had mine replaced by apple.  Of course, I had Fermilab send it to Apple, and it took a bit more than a week to get it back.

Comment (mrguilima) [reply src/physics/base/ImportedProcessAdapter.cc:108]: https://support.apple.com/keyboard-service-program-for-mac-notebooks

Comment (amandalund) [reply src/physics/base/ImportedProcessAdapter.cc]: My delete key has been doubling too, so now I'm getting stuck in lots of infinite loops :(

File: src/physics/base/ImportedProcessAdapter.cc [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        builders[ValueGridType::energy_loss] = ValueGridLogBuilder::from_geant(
+            make_span(vec.x), make_span(vec.y));
+    }
+
+    // Construct range limiters
+    if (ids.range)
+    {
+        const auto& vec = get_vector(ids.range);
+        CELER_ASSERT(vec.vector_type == ImportPhysicsVectorType::log);
+        builders[ValueGridType::range] = ValueGridLogBuilder::from_geant(
```
Comment (stognini) [inline src/physics/base/ImportedProcessAdapter.cc:197]: I see that there is a `ValueGridRangeBuilder` for range tables, but it does not seem to be fully implemented. Is there any particular issue in not using it here? Range tables are also log(E), so probably not, but it was worth mentioning.

File: src/physics/base/ImportedProcessAdapter.hh [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    // Get the table for the given process ID
+    inline const ImportProcess& get(ImportProcessId id) const;
+
+    // Number of imported processes
+    inline ImportProcessId::size_type size() const;
+
+  private:
+    std::vector<ImportProcess> processes_;
+
```
Comment (stognini) [inline src/physics/base/ImportedProcessAdapter.hh:51]: Nitpicking: Maybe remove empty line here.

File: src/physics/base/ImportedProcessAdapter.cc (hunk 3 above)
Comment (sethrj) [reply src/physics/base/ImportedProcessAdapter.cc:197]: Oh good catch -- I'd meant to use it. It's fully implemented but all it does is call the regular log builder with an extra check that the ranges are monotonically increasing.

**Pull request merged**